crates/rethnet_eth/src/remote/client.rs

```diff
@@ -527,20 +594,25 @@ impl RpcClient {
         position: U256,
         block: Option<BlockSpec>,
     ) -> Result<U256, RpcClientError> {
-        self.call(&MethodInvocation::GetStorageAt(*address, position, block))
+        self.call(MethodInvocation::GetStorageAt(*address, position, block))
             .await
     }
 
     /// Calls `net_version`.
     pub async fn network_id(&self) -> Result<U256, RpcClientError> {
-        self.call(&MethodInvocation::NetVersion()).await
+        self.call(MethodInvocation::NetVersion()).await
     }
 }
 
-#[derive(Debug)]
-struct SerializedRequest {
-    id: jsonrpc::Id,
-    request: String,
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[repr(transparent)]
+#[serde(transparent)]
+struct SerializedRequest(serde_json::Value);
```

Comment:
What is the benefit of wrapping?

Reply:
Just to make the purpose of the generic value clear. It has no runtime cost, but it guarantees that we don't accidentally pass a response in place of a request or vice versa.